components/NavBar/Navbar.tsx

```diff
@@ -0,0 +1,95 @@
+import Image from "next/image";
+import Link from "next/link";
+import React, { useState, useEffect } from "react";
+import { AiOutlineMenu } from "react-icons/ai";
+import { useRouter } from "next/router";
+import DrawerNav from "./DrawerNav";
+
+interface NavbarProps {
+  isDark: boolean;
+}
+
+const Navbar: React.FC<NavbarProps> = ({ isDark }) => {
+  const [nav, setNav] = useState<boolean>(false);
+  const [shadow, setShadow] = useState<boolean>(false);
+  const [navBg, setNavBg] = useState<string>("#ecf0f3");
+  const [linkColor, setLinkColor] = useState<string>("#1f2937");
+  const router = useRouter();
+
+  useEffect(() => {
+    if (isDark) {
+      setNavBg("transparent");
+      setLinkColor("#ecf0f3");
+    } else {
+      setNavBg("#ecf0f3");
+      setLinkColor("#1f2937");
+    }
+  }, [isDark]);
+
+  const handleNav = () => {
+    setNav(!nav);
+  };
+
+  const checkScrollTop = () => {
```

Comment:
You might want to wrap your checkScrollTop function with useCallback hook to memoize the function and avoid unnecessary re-renders. It's also important to add checkScrollTop to the dependency array of your useEffect to ensure that the event listener always has the most recent version of the function

```
  const checkScrollTop = useCallback(() => {
    setShadow(window.scrollY >= 90);
  }, []);
  ````